In [1]:
from __future__ import print_function, division

%matplotlib inline

import warnings
warnings.filterwarnings('ignore')

import math
import numpy as np
from scipy.special import gamma

from thinkbayes2 import Pmf, Suite
import thinkplot

# A professor at a large university tries strike up a conversation with a passing student. The professor realizes that the student is shy. Is the student more likely to be in the school of math and sciences, or the business school? 


Having asked several people around campus the most popular answer is that student is probably from the school of masth and science.

First we need our priors:
What do we beleve about meeting a random student's probability of being from the college of math and science?

Well most buisness departments at large institutions are on the order of ten time larger than the math and science departments. So we have a 10% chance of finding a student from the school of math and sciences

In [2]:

buisormat = Pmf (['math','business'])
buisormat['math'] = .1
buisormat['business'] = .9
buisormat.Print()

business 0.9
math 0.1


Now we could either look up statistics that have varying defintions of the word shy to find our probability givin,
or we can just make numbers up for the sake of argument. I choose to do the latter. 
I'll say that about half of all math and science students are shy, and only 10% of business studens are.  

In [3]:
buisormat['math'] *= .5
buisormat['business'] *= .1
buisormat.Normalize()
buisormat.Print()

business 0.6428571428571429
math 0.35714285714285715


So it is actually more likely that the shy student the professor meets is from the buisness school, in fact there is only a 36% chance that the student is from the school of math and science. 

# Two people have left traces of their own blood at the scene of a crime.  A suspect, Oliver, is tested and found to have type O blood.  The blood groups of the two traces are found to be of type O (a common type in the local population, having frequency 60%) and of type AB (a rare type, with frequency 1%).  Do these data (the blood types found at the scene) give evidence in favour [sic] of the proposition that Oliver was one of the two people whose blood was found at the scene?

So our two hypothoses are that A: Oliver's blood was the type O found at the scene and B: His was not the blood at the scene. The evidence E is that there are two blood samples o and AB:

P(A|E) is simply .01 as Oliver's blood accounts for the O and the random probability of AB is .01

P(B|E) is (.01)(.6)(2) = .012 : the two probabilities multiplied and then doubled as you have to draw twice from the population pool. 

Therefore, it is less likely (however slightly) that the blood is not Oliver's


# Elvis Presley had a twin brother who died at birth.  What is the probability that Elvis was an identical twin?

According to the Wikipedia article on twins:  "Twins are estimated to be approximately 1.9% of the world population, with monozygotic twins making up 0.2% of the total---and 8% of all twins."

Hypotheses:
A: Evelis was an identical twin
B: Elvis was a fraternal twin

P(A) = .08
P(B) = .92

E: His BROTHER was male

P(E|A) = 1 because if Elvis had an identical twin, it would be male.
P(E|B) = .5 because a fraternal twin could be male or female. 

P(E) = P(A)P(E|A)+P(B)P(E|B) = .54

P(A|E) = (.08)(1)/(.54) = .148

It is 14.8% probable that Elvis's twin was identical.

# Given the following statistics, what is the probability that a woman, age 51, has cancer if she has a positive mammogram result?



One percent of women over 50 have breast cancer.
Ninety percent of women who have breast cancer test positive on mammograms.
Eight percent of women will have false positives.

Hypotheses:
A: She has cancer
B: She doesn't 
E: She tested positive

P(A) = .01 because that's what the problem said
P(B) = .99 because that's what the problem said

P(E|A) = .9 because that's what the problem said
P(E|B) = .1 because that's what the problem said

P(E) = P(A)P(E|A)+P(B)P(E|B) = .108

P(A|E) = (.01)(.9)/(.108) = .0833

So the probability of a woman having cancer given a possitive test result is only 8.33% which, as a side note, is shocking to me.

